apollo-federation/src/supergraph/subgraph.rs

```diff
@@ -94,6 +99,14 @@ impl ValidFederationSubgraphs {
     pub fn get(&self, name: &str) -> Option<&ValidFederationSubgraph> {
         self.subgraphs.get(name)
     }
+
+    pub(crate) fn upgrade_subgraphs_if_necessary(
+        mut self,
+    ) -> Result<ValidFederationSubgraphs, FederationError> {
+        Ok(ValidFederationSubgraphs {
+            subgraphs: mem::take(&mut self.subgraphs),
+        })
+    }
```

Comment:
What is this method doing?

Reply:
I was starting to change the validation types, but I decided to save that till later, so I removed it for now.